app/controllers/transactions_controller.rb

```diff
@@ -1,63 +1,75 @@
+# frozen_string_literal: true
+
 class TransactionsController < ApplicationController
-  before_action :set_transaction, only: [:edit, :update, :destroy, :show]
+  before_action :authenticate_user!
+  before_action :set_transaction, only: %i[edit update destroy show]
 
   def index
-    @transactions = current_user.transactions.order(created_at: :desc).page(params[:page]).per(5)
+    @sort_order = params[:sort] || 'desc'
```

Comment:
Why you need instance variable?

Reply:
It was not working without the instance variable. The app was keep on breaking. In my POV, the params[:sort] comes from transaction/index.html.erb as a result of clicking on the sort action/link so when there is no click, there is no params[:sort] and resultantly app breaks.